src/EditorFeatures/Core/SolutionEvents/HostLegacySolutionEventsWorkspaceEventListener.cs

```diff
@@ -92,52 +81,13 @@ private async ValueTask ProcessWorkspaceChangeEventsAsync(ImmutableSegmentedList
             }
         }
 
-        private static async ValueTask ProcessEventAsync(ILegacySolutionEventsAggregationService aggregationService, LegacySolutionEvent ev, CancellationToken cancellationToken)
-        {
-            if (ev.DocumentOpenArgs != null)
-            {
-                await aggregationService.OnTextDocumentOpenedAsync(ev.DocumentOpenArgs, cancellationToken).ConfigureAwait(false);
-            }
-            else if (ev.DocumentCloseArgs != null)
-            {
-                await aggregationService.OnTextDocumentOpenedAsync(ev.DocumentCloseArgs, cancellationToken).ConfigureAwait(false);
-            }
-            else
-            {
-                Contract.ThrowIfNull(ev.WorkspaceChangeArgs);
-                await aggregationService.OnWorkspaceChangedAsync(ev.WorkspaceChangeArgs, cancellationToken).ConfigureAwait(false);
-            }
-        }
-
-        private static async ValueTask ProcessEventAsync(RemoteHostClient client, LegacySolutionEvent ev, CancellationToken cancellationToken)
+        private static async ValueTask ProcessEventAsync(RemoteHostClient client, WorkspaceChangeEventArgs args, CancellationToken cancellationToken)
```

Comment:
Would it make sense to inline this too considering that the other overload above was inlined too?

Reply:
yup.  i can do that!